examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingPlayerDiscovery.java

```diff
@@ -0,0 +1,132 @@
+
+import android.util.Log;
+import com.matter.casting.support.MatterError;
+import java.util.List;
+
+/**
+ * The CastingPlayerDiscovery interface defines the API to control Matter Casting Player discovery
+ * over DNS-SD, and to collect discovery results. Discovery is centrally managed by the native C++
+ * layer in the Matter SDK. This class exposes native functions to add and remove a
+ * CastingPlayerChangeListener, which contains the C++ to Java callbacks for when Casting Players
+ * are discovered, updated, or lost from the network. This class is a singleton.
+ */
+public interface CastingPlayerDiscovery {
+
+  /**
+   * @return a list of Casting Players discovered during the current discovery session. This list is
+   *     cleared when discovery stops.
+   */
+  List<CastingPlayer> getCastingPlayers();
+
+  /**
+   * Starts Casting Players discovery or returns an error.
+   *
+   * @param discoveryTargetDeviceType the target device type to be discovered using DNS-SD. 35
+   *     represents device type of Matter Casting Player.
+   * @return a specific MatterError if the the operation failed or NO_ERROR if succeeded.
+   */
+  MatterError startDiscovery(int discoveryTargetDeviceType);
+
+  /**
+   * Stops Casting Players discovery or returns an error.
+   *
+   * @return a specific MatterError if the the operation failed or NO_ERROR if succeeded.
+   */
+  MatterError stopDiscovery();
+
+  /**
+   * Adds a CastingPlayerChangeListener instance to be used during discovery. The
+   * CastingPlayerChangeListener contains the C++ to Java callbacks for when Casting Players are
+   * discovered, updated, or lost from the network. Should be called prior to calling
+   * MatterCastingPlayerDiscovery.startDiscovery().
+   *
+   * @param listener an instance of the CastingPlayerChangeListener to be implemented by the APIs
+   *     consumer.
+   * @return a specific MatterError if the the operation failed or NO_ERROR if succeeded.
+   */
+  MatterError addCastingPlayerChangeListener(CastingPlayerChangeListener listener);
+
+  /**
+   * Removes CastingPlayerChangeListener from the native layer.
+   *
+   * @param listener the specific instance of CastingPlayerChangeListener to be removed.
+   * @return a specific MatterError if the the operation failed or NO_ERROR if succeeded.
+   */
+  MatterError removeCastingPlayerChangeListener(CastingPlayerChangeListener listener);
+
+  /**
+   * The CastingPlayerChangeListener can discover CastingPlayers by implementing the onAdded,
+   * onChanged and onRemoved callbacks which are called as CastingPlayers, are discovered, updated,
+   * or lost from the network. The onAdded(), onChanged() and onRemoved() callbacks must be
+   * implemented by the APIs client.
+   */
+  abstract class CastingPlayerChangeListener {
+    static final String TAG = CastingPlayerChangeListener.class.getSimpleName();
+
+    /**
+     * Called by the native C++ layer when a Casting Player is added to the local network.
```

Comment:
Since the docs are meant for the client, they do not need to know that the implementation involves native C++. We can just state that "this handler is called when a CastingPlayer is added to the network".

Same comment for the docs for the other functions.